homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -56,38 +76,52 @@ def __init__(self, coordinator, description, device):
         self.entity_description = description
 
         self._attr_device_info = device
-        self._attr_should_poll = bool(self.key in ("heat_usage", "heat_previous_year"))
-
-    async def async_added_to_hass(self) -> None:
-        """Call when entity about to be added to hass."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_sensor_data()
-        if state:
-            self._attr_native_value = state.native_value
+        self.update_values_from_api()
```

Comment:
This method should not be needed, the coordinator already holds the data, which can be retrieved from it directly.

Reply:
I'm missing something here: If I remove this statement, the entities get not value. 
Or is the name of the method confusing? All I'm doing in update_values_from_api is get the values from the coordinator. If so, I'll rename it.

Reply:
I've figured this out as well. I've made native_value a property method which does wonders.